project/driver_controls/inc/event_arbiter.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+// Interface for managing raised events
+
+// Each time an event is popped from the event queue, the event will be checked against
+// the current states of all active FSMs to determine whether it will be processed or
+// discarded. This is to prevent situations like processing a gear shift while the brake is
+// not pressed, which would be dangerous for the driver.
+
+#include "fsm.h"
+#include "status.h"
+
+typedef void (*DriverFSMInit)(FSM *fsm, void *context);
```

Comment:
Technically, this doesn't follow the prefixing style rule.

Do you think it would make sense to reverse the dependency? i.e. Instead of needing to pass in an init function and having this run the FSM's init function, have the FSM register itself during init.